frontend/packages/ceph-storage-plugin/src/features.ts

```diff
@@ -20,19 +23,41 @@ export const detectIndependentMode: ActionFeatureFlagDetector = (dispatch) =>
     },
   );
 
-export const OCS_VERSION_4_5_FLAG = 'OCS_VERSION_4_5_FLAG';
+export const OCS_SUPPORT_FLAGS = {
+  OCS_SNAPSHOT: 'OCS_SNAPSHOT',
+  // OCS_EXTERNAL: 'OCS_EXTERNAL'
+};
+
+const setDefaultFlag = (dispatch: Dispatch<AnyAction>) => {
+  _.keys(OCS_SUPPORT_FLAGS).forEach((feature) => {
+    dispatch(setFlag(feature, false));
+  });
+};
 
-export const isOCS45AndAboveVersion = (subscription: K8sResourceKind): boolean => {
-  const version = subscription?.status?.currentCSV;
-  return version && version.includes('ocs-operator.v4') && version.split('.')[2] >= 5;
+const handleFlagError = (err, dispatch: Dispatch<AnyAction>) => {
+  _.keys(OCS_SUPPORT_FLAGS).forEach((feature) => {
+    handleError(err, feature, dispatch, detectOCSSupportedFeatures);
+  });
 };
 
-export const detectOCSVersion45AndAbove: ActionFeatureFlagDetector = (dispatch) =>
+export const detectOCSSupportedFeatures: ActionFeatureFlagDetector = (dispatch) =>
   k8sGet(SubscriptionModel, 'ocs-subscription', CEPH_STORAGE_NAMESPACE).then(
-    (obj: K8sResourceKind) => dispatch(setFlag(OCS_VERSION_4_5_FLAG, isOCS45AndAboveVersion(obj))),
-    (err) => {
-      err?.response?.status === 404
-        ? dispatch(setFlag(OCS_VERSION_4_5_FLAG, false))
-        : handleError(err, OCS_VERSION_4_5_FLAG, dispatch, detectOCSVersion45AndAbove);
+    (subscription: K8sResourceKind) => {
+      // eslint-disable-line promise/no-nesting
+      k8sGet(
+        ClusterServiceVersionModel,
+        subscription?.status?.currentCSV,
+        CEPH_STORAGE_NAMESPACE,
+      ).then(
+        (obj: K8sResourceKind) => {
+          const support = obj?.metadata?.annotations['features.ocs.openshift.io/enabled'];
+          _.keys(OCS_SUPPORT_FLAGS).forEach((feature) => {
+            dispatch(setFlag(feature, support.indexOf(feature.split('_')[1].toLowerCase()) !== -1));
```

Comment:
```suggestion
            dispatch(setFlag(feature, support.includes(feature))));
```
We need to ask OCS to send the flags as we want them.